app/addons/documents/shared-resources.js

```diff
@@ -0,0 +1,317 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  'app',
+  'api',
+  'cloudant.pagingcollection'
+], function (app, FauxtonAPI, PagingCollection) {
+
+  var Documents = FauxtonAPI.addon();
```

Comment:
@benkeen thats fine. Define the addon in shared. I think leave a comment there so we know why its defined there.